### PR TITLE
Module 4

### PR DESCRIPTION
**Pull request by a-tarasiuk**

### CODE REVIEW

File: module_4/gift_certificates_system/controller/src/main/java/com/epam/esm/controller/controller/GiftCertificateController.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        return EntityModel.of(certificate);
+    }
+
+    /**
+     * Delete gift certificate by it is ID.
+     *
+     * @param id Gift certificate ID.
+     */
+    @DeleteMapping(path = UrlMapping.ID)
+    public ResponseEntity<Void> delete(@PathVariable long id) {
```
Comment (labutss) [inline module_4/gift_certificates_system/controller/src/main/java/com/epam/esm/controller/controller/GiftCertificateController.java]: Please return void + @ResponseStatus

File: module_4/gift_certificates_system/controller/src/main/java/com/epam/esm/controller/filter/JwtRequestFilter.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+@Component
+@Log4j2
+public class JwtRequestFilter extends OncePerRequestFilter {
+
+    @Autowired
+    private UserService userService;
+
+    @Autowired
+    private JwtUtils jwtUtils;
```
Comment (labutss) [inline module_4/gift_certificates_system/controller/src/main/java/com/epam/esm/controller/filter/JwtRequestFilter.java]: please avoid field injection

File: module_4/gift_certificates_system/controller/src/main/java/com/epam/esm/controller/util/PackageName.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package com.epam.esm.controller.util;
+
+public final class PackageName {
```
Comment (labutss) [inline module_4/gift_certificates_system/controller/src/main/java/com/epam/esm/controller/util/PackageName.java]: Please remove

File: module_4/gift_certificates_system/model/src/main/java/com/epam/esm/model/util/DtoRegexValidator.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public final class DtoRegexValidator {
+    // User DTO
+    public static final String USER_LOGIN = "^[\\p{Ll}\\p{Lu}]{4,50}$";
+    public static final String USER_NAME = "^\\p{Lu}\\p{Ll}{1,50}$";
+
+    // Tag DTO
+    public static final String TAG_NAME = "^\\p{Lu}\\p{Ll}{1,50}$";
+
+    // Gift certificate DTO
```
Comment (labutss) [inline module_4/gift_certificates_system/model/src/main/java/com/epam/esm/model/util/DtoRegexValidator.java]: Please remove comments, everything is clear from the naming

File: module_4/gift_certificates_system/model/src/main/java/com/epam/esm/model/util/UrlMapping.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+    /**
+     * URL for Order
+     */
+    public static final String ORDERS = "/orders";
+    public static final String FIND_ORDERS_BY_ID = "/orders/{id}";
+
+    /**
+     * URL for Authentication
+     */
```
Comment (labutss) [inline module_4/gift_certificates_system/model/src/main/java/com/epam/esm/model/util/UrlMapping.java]: Please remove comments

File: module_4/gift_certificates_system/model/src/main/java/com/epam/esm/model/entity/Order.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+import static com.epam.esm.model.util.DatabaseColumnName.GIFT_CERTIFICATE_ID;
+import static com.epam.esm.model.util.DatabaseColumnName.USER_ID;
+
+/**
+ * Order entity.
+ *
+ * @see com.epam.esm.model.listener.OrderListener
+ */
+@RequiredArgsConstructor
```
Comment (labutss) [inline module_4/gift_certificates_system/model/src/main/java/com/epam/esm/model/entity/Order.java]: can't see required parameters
please remove

File: module_4/gift_certificates_system/repository/src/test/java/com/epam/esm/repository/configuration/RepositoryConfigurationTest.java [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+/**
+ * Main ESM application configuration.
+ */
+@Configuration
+@EnableTransactionManagement
+@EnableJpaRepositories("com.epam.esm")
+@ComponentScan("com.epam.esm")
+@EntityScan("com.epam.esm")
+@SpringBootConfiguration
+public class RepositoryConfigurationTest implements WebMvcConfigurer {
```
Comment (labutss) [inline module_4/gift_certificates_system/repository/src/test/java/com/epam/esm/repository/configuration/RepositoryConfigurationTest.java]: Incorrect naming, looks as if it is test for configuration, not test configuration

File: module_4/gift_certificates_system/model/src/main/java/com/epam/esm/model/util/DtoRegexValidator.java (hunk 4 above)
Comment (a-tarasiuk) [reply module_4/gift_certificates_system/model/src/main/java/com/epam/esm/model/util/DtoRegexValidator.java]: Thanks, was [removed](https://github.com/n199a/epam_training_center/pull/4/commits/8d4ce0b980d1aea206cf45781d79f362b3581b15).

File: module_4/gift_certificates_system/model/src/main/java/com/epam/esm/model/util/UrlMapping.java (hunk 5 above)
Comment (a-tarasiuk) [reply module_4/gift_certificates_system/model/src/main/java/com/epam/esm/model/util/UrlMapping.java]: Thanks, was [removed](https://github.com/n199a/epam_training_center/pull/4/commits/be45a9462502ed1fbd50dd25761f60ef27ecbe0b).

File: module_4/gift_certificates_system/model/src/main/java/com/epam/esm/model/entity/Order.java (hunk 6 above)
Comment (a-tarasiuk) [reply module_4/gift_certificates_system/model/src/main/java/com/epam/esm/model/entity/Order.java]: Thanks, was removed.

File: module_4/gift_certificates_system/controller/src/main/java/com/epam/esm/controller/controller/GiftCertificateController.java (hunk 1 above)
Comment (a-tarasiuk) [reply module_4/gift_certificates_system/controller/src/main/java/com/epam/esm/controller/controller/GiftCertificateController.java]: This method `delete(long id)` used in the `GiftCertificateLinkBuilder.class` to create links for `HATEOAS`. Overload method `org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.linkTo` cannot take a type argument `void`.

File: module_4/gift_certificates_system/repository/src/test/java/com/epam/esm/repository/configuration/RepositoryConfigurationTest.java (hunk 7 above)
Comment (a-tarasiuk) [reply module_4/gift_certificates_system/repository/src/test/java/com/epam/esm/repository/configuration/RepositoryConfigurationTest.java]: Thanks, was [renamed](https://github.com/n199a/epam_training_center/pull/4/commits/7284879fec13d1ad2e6683c55327916e6fbad158).

File: module_4/gift_certificates_system/controller/src/main/java/com/epam/esm/controller/filter/JwtRequestFilter.java (hunk 2 above)
Comment (a-tarasiuk) [reply module_4/gift_certificates_system/controller/src/main/java/com/epam/esm/controller/filter/JwtRequestFilter.java]: Thanks, [resolved](https://github.com/n199a/epam_training_center/pull/4/commits/ef4d100c6b85c1e2519ae112ce72c2bbafa61aeb).

File: module_4/gift_certificates_system/controller/src/main/java/com/epam/esm/controller/util/PackageName.java (hunk 3 above)
Comment (a-tarasiuk) [reply module_4/gift_certificates_system/controller/src/main/java/com/epam/esm/controller/util/PackageName.java]: `PackageName.class` was renamed to `PackagePath.class` and used in the `EsmApplicationRunner.class`

**Pull request merged**